crates/primitives/src/attr.rs

```diff
@@ -1,5 +1,59 @@
-use bitflags::bitflags;
+//! Constraints and Column attributes.
+//!
+//! ## For tables
+//!
+//! The database engine support a sub-set of constraints defined by the `SQL` standard:
+//!
+//! ### UNIQUE:
+//!
+//!  One or many columns with enforced uniqueness using an auto-generate index.
+//!
+//! ### IDENTITY:
+//!
+//! A column that is made up of values generated by the database using a sequence.
+//!
+//!  Differs from `PRIMARY_KEY` in that its values are managed by the database and usually cannot be modified.
+//!
+//! ### PRIMARY_KEY:
+//!
+//! One or many columns that uniquely identifies each record in a table.
+//!
+//! Enforce uniqueness using an auto-generate index.
+//!
+//! Can only be one per-table.
+//!
+//! ## For Columns
+//!
+//! Additionally, is possible to add markers to columns as:
+//!
+//! - AUTO_INC: Auto-generate a sequence
+//! - INDEXED: Auto-generate a non-unique index
+//! - PRIMARY_KEY_AUTO: Make it a `PRIMARY_KEY` + `AUTO_INC`
+//! - PRIMARY_KEY_IDENTITY: Make it a `PRIMARY_KEY` + `IDENTITY`
+//!
+//! NOTE: We have [ConstraintKind] and [AttributeKind] intentionally semi-duplicated because
+//! the first is for the [Constrains] that are per-table and the second is for markers of the column.
 
+/// The assigned constraint for a `Table`
+#[allow(non_camel_case_types)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord)]
```

Comment:
IMO these are all confused. We should just have attributes which are separate from constraints:

IMO these shouldn't even be on the same axis.

We have the following column attributes:
- auto_inc

We should have the following table attributes:
- primary_key e.g. `(col_a, col_b)`
- indexed e.g. `(col_a, col_b)`

We should have the following table constraints:
- unique e.g. `(col_a, col_b)`

These are all mutually exclusive and should not be combined into a single "kind"

Also constraints != attributes!

Reply:
This will be addressed in a separate PR, this one is to solve the immediate issue with deploy.